docker/gentoo.Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Run on Gentoo
+#
+# docker build --progress=plain -t wild-dev-gentoo . -f docker/gentoo.Dockerfile
+# docker run -it wild-dev-gentoo
+
+FROM gentoo/stage3:desktop-20251208 AS chef
+
+RUN echo "dev-util/rustup" >> /etc/portage/package.accept_keywords/wild
+RUN emerge-webrsync
+# Required for installing binary packages.
+RUN getuto
+RUN emerge -g dev-util/rustup llvm-core/clang llvm-core/lld
+# Ensures the PATH is correct upon running the image.
+RUN echo 'export PATH="/root/.cargo/bin:${PATH}"' >> /etc/profile
+
+RUN rustup-init --default-toolchain stable -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+RUN rustup toolchain install nightly && \
```

Comment:
This is how most of our docker files install the toolchain, but recently we've started using the following:

```docker
RUN rustup toolchain install nightly \
        --allow-downgrade \
        --target x86_64-unknown-linux-musl,aarch64-unknown-linux-gnu,aarch64-unknown-linux-musl,riscv64gc-unknown-linux-gnu,riscv64gc-unknown-linux-musl \
        --component rustc-codegen-cranelift-preview
```

This allows us to use a slightly older nightly if not all of the components are available rather than failing outright. I think we're possibly only using it in `arch.Dockerfile` and maybe one other so far, but it could be good to use here.